### PR TITLE
Disable smart-limit by default

### PR DESCRIPTION
**Pull request by li-boxuan**
Prior to 0.6.0, `smart-limit` is enabled by default. It tries to guess a small limit
for each query internally, and if more results are required by user, it queries
backend again with a larger limit, and repeats until either results are exhausted or
user stops the query. However, this is not the same as paging mechanism. All interim
results will be fetched again in next round, making the whole query costly. Until
janusgraph can fully utilize the paging capacity provided by backends (e.g. Elasticsearch
scroll), I am inclined to make this option disabled by default.

Related issue: https://github.com/JanusGraph/janusgraph/issues/735

Any thoughts? Do you enable this option in your production environment?

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java [hunk 1]
```diff
@@ -259,7 +259,7 @@
     public static final ConfigOption<Boolean> ADJUST_LIMIT = new ConfigOption<>(QUERY_NS,"smart-limit",
             "Whether the query optimizer should try to guess a smart limit for the query to ensure responsiveness in " +
                     "light of possibly large result sets. Those will be loaded incrementally if this option is enabled.",
-            ConfigOption.Type.MASKABLE, true);
+            ConfigOption.Type.MASKABLE, false);
```
Comment (porunov) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java:262]: After updating configuration options we need to run `mvn --quiet clean install -DskipTests=true -pl janusgraph-doc -am` to regenerate configuration references documentation.
Probably we need to make it more clear

**Comment by porunov** [comment]
I actually didn't know that smart limit queries the same data over and over again. Does it doubles `limit` each time or increases `limit` somehow differently? If it doubles then in some cases, the amount of data returned will be n + n\2 + n\4 + n\8... I.e. 2x overhead which isn't nice.
That said, it's interesting why this smart limit was used by default. I.e. in what cases it actually improves performance. I think it has to have some logic behind such behavior. That said, I didn't check the logic yet

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java (hunk 1 above)
Comment (li-boxuan) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java:262]: My bad, forgot about that. Thanks for reminder.

**Comment by li-boxuan** [comment]
> Does it doubles limit each time or increases limit somehow differently?

It doubles the limit each time. See the code and doc in LimitAdjustingIterator:

https://github.com/JanusGraph/janusgraph/blob/3787fd4d78d2aa5e849ea836852901b19a469ec5/janusgraph-core/src/main/java/org/janusgraph/graphdb/query/LimitAdjustingIterator.java#L23-L35

and (one of its) implementation class:

https://github.com/JanusGraph/janusgraph/blob/3787fd4d78d2aa5e849ea836852901b19a469ec5/janusgraph-core/src/main/java/org/janusgraph/graphdb/query/QueryProcessor.java#L177-L184

Ideally, this could act like a paging mechanism, so that the next query does not have to fetch every result from the very beginning, but from the cursor. However, it does not work like this. To properly utilize this smart-limit option and LimitAdjustingIterator, we need to:
1) Support paging in this iterator so that it does not fetch from beginning
1) Storage adapter must support paging. Elasticsearch supports paging, but I am not sure about other storage we use (especially primary storage backends).

Now, to my best knowledge, the only benefit of this smart-limit thing is when you are using native java API and you are iterating the results like this:

```java
Iterator iter = g.V();
int count = 0;
while (iter.hasNext()) {
    if (iter.next() matches some condition) break;
}
```
However, even the above use case can be implemented manually without the "smart-limit" support since this "smart-limit" mechanism is very naive.

**Comment by li-boxuan** [comment]
Also it is worth noting that even if smart-limit is disabled, this "LimitAdjustingIterator" curse is still there because of 

https://github.com/JanusGraph/janusgraph/blob/3787fd4d78d2aa5e849ea836852901b19a469ec5/janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/GraphCentricQueryBuilder.java#L364-L368

However, this HARD_MAX_LIMIT is 100000 (compared to DEFAULT_NO_LIMIT == 1000 when smart-limit is enabled), which makes the situation slightly better compared to the case where smart-limit is enabled.

**Review by porunov — APPROVED** [review]
LGTM. Thank you for the explanation @li-boxuan !

**Pull request merged**